src/rpc/rawtransaction.cpp

```diff
@@ -1669,7 +1669,7 @@ UniValue converttopsbt(const JSONRPCRequest& request)
 
     // Remove all scriptSigs and scriptWitnesses from inputs
     for (CTxIn& input : tx.vin) {
-        if ((!input.scriptSig.empty() || !input.scriptWitness.IsNull()) && (request.params[1].isNull() || (!request.params[1].isNull() && request.params[1].get_bool()))) {
+        if ((!input.scriptSig.empty() || !input.scriptWitness.IsNull()) && !permitsigdata) {
```

Comment:
Would it make sense to have the `!request.params[1].isNull()` in line 1661 above?

Reply:
Why? `permitsigdata` is enough?